### PR TITLE
Homework 1: Iris 

### PR DESCRIPTION
**Pull request by timand32**
0. As a heads up, you may need to recompile the requirements.dev.txt file from the requirements.dev.in file in order for pip to work.  I think we worked with different versions of Python, as I was getting errors trying to use pip to run your code with your requirements.dev.txt file.
1. apps/iris/iris.py is the main executable Python file, the rest are modules for it.  I realize now I should have added a hw1.py file to make it clear what file is my submission.  If that seems like a good idea, I can add it.

### CODE REVIEW

File: apps/iris/iris.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    iris_df = data.load_data()
+    print(iris_df)
+    iris_desc = data.describe_data(iris_df)
+    print(iris_desc)
+    iris_plots = plots.generate_plots(iris_df)
+    for plot in iris_plots:
+        plot.show(validate=False)
+    pipelines = models.build_pipelines()
+    pipelines = models.train_pipelines(iris_df, pipelines)
+    models.analyze_pipelines(pipelines)
```
Comment (amypetris) [inline apps/iris/iris.py]: I ran this file and got a  `TypeError: analyze_pipelines() missing 1 required positional argument: 'pipelines' `error. models.analyze_pipelines takes in 2 arguments: iris_df and pipelines. Passing the dataframe fixes this error.

File: apps/iris/plots.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        figure.add_trace(trace, 1, 1)
+    for trace in petal_plot.data:
+        figure.add_trace(trace, 1, 2)
+
+    # Stack exchange 58849925 helped here understanding editing Figures.
+    figure.layout["xaxis"]["title"] = "sepal width (cm)"
+    figure.layout["yaxis"]["title"] = "sepal length (cm)"
+    figure.layout["xaxis2"]["title"] = "petal width (cm)"
+    figure.layout["yaxis2"]["title"] = "petal length (cm)"
+    figure.layout["title"] = "Scatterplots of Iris Features"
```
Comment (amypetris) [inline apps/iris/plots.py:45]: Legend has duplicate entries for each subplot. I saw some people using go.Scatter with the showlegend=False option. Maybe that will work, if you have not already tried it.

**Review by amypetris — COMMENTED** [review]
Checked out the branch and ran iris.py. Was getting an error with analyze_pipelines(see comment). Once I fixed this error, everything ran smoothly. I also left a comment about duplicate legend entries in some of the plots. Other than that, everything looks really good. Great class structure and function comments.

File: apps/iris/iris.py (hunk 1 above)
Comment (timand32) [reply apps/iris/iris.py]: That's a pretty big thing for me to miss, thank you for catching this!  This should be fixed in `db9eb87`.

File: apps/iris/plots.py (hunk 2 above)
Comment (timand32) [reply apps/iris/plots.py:45]: Thanks for the suggestion, I was able to remove the duplicate entries with this in `051a889`.

**Pull request merged**